server/utils/aws/aws_sts_client.py

```
@@ -123,8 +125,8 @@ def assume_workspace_role(
         
         if external_id != workspace_external_id:
             logger.error(
-                f"SECURITY: External ID mismatch for workspace {workspace_id}. "
-                f"Provided: '{external_id}', Expected: '{workspace_external_id}'. Rejecting role assumption."
+                f"SECURITY: External ID mismatch for workspace {sanitize(workspace_id)}. "
+                f"Provided: '{sanitize(external_id)}', Expected: '{sanitize(workspace_external_id)}'. Rejecting role assumption."
```

Ruff (0.15.11)

[warning] Logging statement uses f-string

(G004)